src/main/java/net/neoforged/neoforge/attachment/IAttachmentHolder.java

```diff
@@ -46,6 +47,22 @@ default <T> T getData(Supplier<AttachmentType<T>> type) {
         return getData(type.get());
     }
 
+    /**
+     * {@return an optional possibly containing a data attachment value of the given type}
+     *
+     * <p>If there is no data attachment of the given type, an empty optional is returned.
+     */
+    <T> Optional<T> getOptionalData(AttachmentType<T> type);
```

Comment:
I would like to call this method something like `getExistingData`, so that it is immediately obvious why the data is optional.

Reply:
Hm. I don't have any problems with that name, so I'll change it when I'm able. Til then, don't merge